lib/dynamode/storage/index.ts

```diff
@@ -138,28 +159,25 @@ export default class DynamodeStorage {
     const metadata = this.getEntityMetadata(entityName);
     const attributes = this.getEntityAttributes(entityName);
 
-    // Validate metadata
-    validateMetadataUniqueness(entityName, metadata);
-
     // Validate decorated attributes
     Object.entries(attributes).forEach(([name, attribute]) =>
       validateDecoratedAttribute({ name, attribute, metadata, entityName }),
     );
 
     // Validate table partition key
-    validateMetadataAttribute({ name: metadata.partitionKey, attributes, role: 'partitionKey', entityName });
+    validateMetadataAttribute({ name: metadata.partitionKey, attributes, validRoles: ['partitionKey'], entityName });
```

Comment:
Why do you need to enforce that? 

It shouldn't be the case - e.g., you can use the main table *sort key* as a GSI *partition key*.

Reply:
This was done to ensure that decorated keys from the entity matches these defined in Table's metadata (second argument of `TableManager` constructor). 

The only valid role for primary partition key is 'partitionKey', that's why it's provided

Reply:
Oh yeah, indeed! In this instance, it makes perfect sense - as in the latest commit, later in the code, the GSI part missed the additional "role" in validation.